### ISSUE TITLE
Zero windspeed on zero meridian

### ISSUE THREAD

**Issue by SanderHulst** [NOAA-EMC/WW3#242]
The _averaging_ code in WAVEWATCH appears not to properly project 0.125 degree ECMWF surface wind data on our global 30 minute grid, while 0.25 degree GFS data seems to be projected to the target grid just fine. An important aspect to keep in mind is that both grids have their left origin on the zero meridian. Would they have been on -180, the problem would manifest itself there.

A bug is present that affected both datasources but only results in the zero windspeed in the ECMWF case because of a threshold in the same averaging code.

It was traced to wmupdmd.ftn:
```
          DO J=IXS0, IXSN
              JJ=J
              IF ( FLAGLL ) THEN
                IF ( ICLOSE.NE.ICLOSE_NONE ) JJ = 1 + MOD(J-1,NXI)
                IF ( JJ.LT.1 .OR. JJ.GT. NXI ) CYCLE
<compute weight of source grid cell>
           END DO 
           IF ( NXA(IX,0)  .GT. 0 ) THEN
              RT     = SUM ( RXA(IX,:) )
              IF ( RT .LT. 0.7 ) THEN <---- NOTE THE THRESHOLD
                NXA(IX,:) = 0
                RXA(IX,:) = 0.
              END IF
            END IF
          END DO
```
J is the index of the source grid cell considered. The MOD is used to fold the right to the left side of the source grid, but it does not fold the left to the right because negative indices pass the mod statement as is. In the case of a negative index or zero, the CYCLE will cause the source point not to be considered. Below the output of a [test program](https://github.com/NOAA-EMC/WW3/files/5062046/test_fortran_stuff.txt) is printed that simulates 0.25 degree source data and a half a degree target.

```
X      : 0.000
SX     : 0.500
SXI    : 0.250
NXI    :  1440
MXA    :     4
IX     :   361
XFL    : -.250
XFR    : 0.250
IXC    :     1
IXS0   :    -2
IXSN   :     4
Skipped JJ    -2
Skipped JJ    -1
Skipped JJ     0
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    1   0   -0.250    0.250   -0.12500    0.12500    0.500
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    2   0   -0.250    0.250    0.12500    0.25000    0.250
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    3   0   -0.250    0.250    0.37500    0.25000    0.000
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    4   0   -0.250    0.250    0.62500    0.25000    0.000
Average over  2 points with a total weight of  0.750
```
The interesting bit is that the R1 weights add up to 0.75, above the threshold of RT < 0.7. So, eventhough the result is not as intented, it at least has a value on the zero meridian.

In case of the higher 1/8th of a degree resolution you get:
```
Skipped JJ    -3
Skipped JJ    -2
Skipped JJ    -1
Skipped JJ     0
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    1   0   -0.250    0.250   -0.06250    0.06250    0.250
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    2   0   -0.250    0.250    0.06250    0.18750    0.250
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    3   0   -0.250    0.250    0.18750    0.25000    0.125
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    4   0   -0.250    0.250    0.31250    0.25000    0.000
JJ, TOGGLED, XFL, XFR, XSL, XSR, R1   :    5   0   -0.250    0.250    0.43750    0.25000    0.000
 All weights set to zero
Average over  0 points with a total weight of  0.625
```
..because 0.625 is obviously < 0.7.

The fix is to simply change
```
IF ( ICLOSE.NE.ICLOSE_NONE ) JJ = 1 + MOD(J-1,NXI)
```
into
```
IF ( ICLOSE.NE.ICLOSE_NONE ) JJ = 1 + MOD(J-1+NXI,NXI)
```

![zero_meridian_gfs_small](https://user-images.githubusercontent.com/37443533/90004798-8dc62680-dc96-11ea-973f-a6a199f46fc4.png)
averaged GFS data
![zero_meridian_ecmwf_global_30min_small](https://user-images.githubusercontent.com/37443533/90004801-8e5ebd00-dc96-11ea-8e15-af1d426e1976.png)
averaged ECMWF data
![windwave_significant_waveheight_small](https://user-images.githubusercontent.com/37443533/90004803-8ef75380-dc96-11ea-9e83-dc2da804b52e.png)
windwaves computed with averaged ECMWF data

**Comment by aliabdolali** [comment]
Hi @SanderHulst What about this one? I will work on this one as well once you confirm.
AA

**Comment by SanderHulst** [comment]
Confirmed. We run a patched v6.07.1 (which we call v6.07.3) operationally forced by ECMWF wind. Works fine.
